contracts/domainsSetLibrary.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.18;
+
+library domainSet {
+    struct Domain {
+        string name;
+        address owner;
+        uint256 deposit;
+    }
+
+    struct Set {
+        Domain[] _values;
+        mapping(string => uint256) _positions;
+    }
```

Comment:
As I prompted before, you could use this 
```
struct Domain {        // note, you don't even need to store domain name here 
    address owner;
    uint256 deposit;
}
mapping (string => Domain);
```
That would be much simpler, more gas efficient and simplified the code that you have in your libraries.
       -2 points